cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -228,6 +233,35 @@ struct MinMaxState<ArrowType, SimdLevel, enable_if_floating_point<ArrowType>> {
   bool has_values = false;
 };
 
+template <typename ArrowType, SimdLevel::type SimdLevel>
+struct MinMaxState<ArrowType, SimdLevel, enable_if_decimal<ArrowType>> {
+  using ThisType = MinMaxState<ArrowType, SimdLevel>;
+  using T = typename std::conditional<is_decimal128_type<ArrowType>::value, Decimal128,
+                                      Decimal256>::type;
```

Comment:
Hmm... perhaps we can add something to `type_traits.h`? I don't know if it should be `TypeTraits<Decimal128>::CType` or some other trait. cc @bkietz

Reply:
TypeTraits<Decimal128>::CType is already defined, so I changed to use that.